src/test/java/org/testng/xml/XmlSuiteTest.java

```diff
@@ -42,13 +44,74 @@ public void testTimeOut(String timeout, int size, int lineNumber) throws IOExcep
         StringReader stringReader = new StringReader(suite.toXml());
         List<String> resultLines = Lists.newArrayList();
         List<Integer> lineNumbers = grep(stringReader, "time-out=\"1000\"", resultLines);
-        assertEquals(lineNumbers.size(), size);
-        assertEquals(resultLines.size(), size);
+        assertThat(lineNumbers.size()).isEqualTo(size);
+        assertThat(resultLines.size()).isEqualTo(size);
         if (size > 0) {
-            assertEquals(lineNumbers.get(size - 1).intValue(), lineNumber);
+            assertThat(lineNumbers.get(size-1)).isEqualTo(lineNumber);
         }
     }
 
+    @Test(expectedExceptions = TestNGException.class,
+    expectedExceptionsMessageRegExp = "\nTwo tests in the same suite \\[random_suite\\] cannot have the same name: test1")
+    public void testEnsureNoDuplicateTestsArePresentNegativeCondition() {
+        XmlSuite suite = createDummySuiteWithTestNamesAs("test1", "test1");
+        suite.ensureNoDuplicateTestsArePresent();
+    }
+
+    @Test
+    public void testEnsureNoDuplicateTestsArePresent() {
+        XmlSuite suite = createDummySuiteWithTestNamesAs("test1", "test2");
+        suite.ensureNoDuplicateTestsArePresent();
+    }
+
+    @Test
+    public void testCloneIfContainsTestsWithNamesMatchingAny() {
+        XmlSuite suite = createDummySuiteWithTestNamesAs("test1", "test2");
+        XmlSuite clonedSuite = suite.cloneIfContainsTestsWithNamesMatchingAny(Collections.singletonList("test2"));
+        assertThat(suite.getTests()).hasSameElementsAs(clonedSuite.getTests());
+    }
+
+    @Test(description = "GITHUB-1594", dataProvider = "getTestnames")
+    public void testCloneIfContainsTestsWithNamesMatchingAnyChildSuites(String testname) {
+        XmlSuite suite = createDummySuiteWithTestNamesAs("test1", "test2");
+        suite.setName("parent_suite");
+        XmlSuite childSuite = createDummySuiteWithTestNamesAs("test3", "test4");
+        childSuite.setName("child_suite");
+        suite.getChildSuites().add(childSuite);
+        XmlSuite clonedSuite = suite.cloneIfContainsTestsWithNamesMatchingAny(Collections.singletonList(testname));
+        assertThat(suite.getTests()).hasSameElementsAs(clonedSuite.getTests());
+    }
+
+    @Test(expectedExceptions = TestNGException.class,
+            expectedExceptionsMessageRegExp = "\nPlease provide a valid list of names to check.",
+            dataProvider = "getData")
+    public void testCloneIfContainsTestsWithNamesMatchingAnyNegativeCondition(XmlSuite xmlSuite, List<String> names) {
+        xmlSuite.cloneIfContainsTestsWithNamesMatchingAny(names);
+    }
+
+    @Test(expectedExceptions = TestNGException.class,
+    expectedExceptionsMessageRegExp = "\nThe test\\(s\\) \\<\\[test3\\]\\> cannot be found.")
```

Comment:
Have applied formatting from the IDE and still this is how it is. So i guess the indentation is proper.

Reply:
Indentation